presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/presto/etc/catalog/kafka/write_all_datatypes_json.json

```diff
@@ -109,18 +109,6 @@
         "dataFormat": "custom-date-time",
         "formatHint": "mm:HH:ss"
       },
-      {
-        "name": "c_timestamptz_milliseconds_since_epoch",
-        "type": "TIMESTAMP WITH TIME ZONE",
-        "mapping": "j_timestamptz_milliseconds_since_epoch",
-        "dataFormat": "milliseconds-since-epoch"
-      },
-      {
-        "name": "c_timestamptz_seconds_since_epoch",
-        "type": "TIMESTAMP WITH TIME ZONE",
-        "mapping": "j_timestamptz_seconds_since_epoch",
-        "dataFormat": "seconds-since-epoch"
```

Comment:
Why is it being deleted?
maybe just don't add it in the first place?

Reply:
This isn't expected. I'll check. Thanks for flagging.

Reply:
Ok. I removed these because the encoder doesn't support writing TIMESTAMP WITH TIME ZONE using the millis-since-epoch or seconds-since-epoch JSON formatters.

But there's no reason why these shouldn't work since the decoder supports reading such values by hardcoding them to UTC. I'll make the change and test again.

Reply:
I think we consciously decided that for formats which does not have notion of timezone, writing `TIMESTAMP/wTZ` should not  be supported. So the information is not lost.
The motivation for decision to have asymmetry between read and write flow, was that there is a difference between coming up with default timezone during read (which is harmless) and loosing explicit important information on write.

Reply:
I'll add this as a comment to the place where the check is enforced.

I can see that there are some conversions that are irreversible/ambiguous and can understand why that decision was made.